### ISSUE TITLE
(not-SAB) hibernate on R540 with Ubuntu 17.10

### ISSUE THREAD

**Issue by sanderjo** [sabnzbd/sabnzbd#948]
Pinging @BrendanBall

Computer #1:
Samsung R540 laptop, i3 core, SSD drive, 4GB RAM
Ubuntu 17.10
swap on separate partition
```
$ swapon
NAME      TYPE      SIZE USED PRIO
/dev/sda5 partition 4,2G 1,6G   -1
```

`systemctl hibernate` as user gives a GUI popup asking for my password. After filling it out, the system actually goes into hibernate. So bad resp good.
CLI says:

```
sander@R540:~$ systemctl hibernate
Failed to hibernate system via logind: Access denied
A dependency job for hibernate.target failed. See 'journalctl -xe' for details.
sander@R540:~$ 
```

Your PR code does nothing:

```
sander@R540:~/git/BrendanBall-sab$ time python -c "import sabnzbd.powersup; print sabnzbd.powersup.linux_hibernate()"
None

real	0m10,752s
user	0m0,492s
sys	0m0,088s
sander@R540:~/git/BrendanBall-sab$ 
```
It probably hangs on a hidden authorisation?

sudo does not solve it:
```
sander@R540:~/git/BrendanBall-sab$ sudo  python -c "import sabnzbd.powersup; print sabnzbd.powersup.linux_hibernate()"
None
sander@R540:~/git/BrendanBall-sab$ 
```

On a next try, I get this:
```
$ systemctl hibernate
Operation inhibited by "sander" (PID 1234 "gnome-session-b", user sander), reason is "user session inhibited".
Please retry operation after closing inhibitors and logging out other users.
Alternatively, ignore inhibitors and users with 'systemctl hibernate -i'.
```
A `systemctl hibernate -i` still works.

I will now reboot to see what happens.

**Comment by sanderjo** [comment]
Hmmm ... `Authentication is required for hibernating the system` is not what I want.

```
sander@R540:~$ pkaction --verbose --action-id org.freedesktop.login1.hibernate
org.freedesktop.login1.hibernate:
  description:       Hibernate the system
  message:           Authentication is required for hibernating the system.
  vendor:            The systemd Project
  vendor_url:        http://www.freedesktop.org/wiki/Software/systemd
  icon:              
  implicit any:      auth_admin_keep
  implicit inactive: auth_admin_keep
  implicit active:   yes

sander@R540:~$ 
```

**Comment by BrendanBall** [comment]
mine says the same although it works and hasn't actually asked for authentication
```
➜  ~ pkaction --verbose --action-id org.freedesktop.login1.hibernate
org.freedesktop.login1.hibernate:
  description:       Hibernate the system
  message:           Authentication is required for hibernating the system.
  vendor:            The systemd Project
  vendor_url:        http://www.freedesktop.org/wiki/Software/systemd
  icon:              
  implicit any:      auth_admin_keep
  implicit inactive: auth_admin_keep
  implicit active:   yes
```
I might have to install ubuntu to see why it's different. I could understand something like this in actual sabnzbd if it's running as a service as a different user, but not just running the pytthon code like this.

**Comment by BrendanBall** [comment]
@sanderjo So I installed ubuntu 17.04 and testing hibernate. It seems that it does require authentication by default. https://wiki.archlinux.org/index.php/systemd#Power_management says 
```
If you are in a local systemd-logind user session and no other session is active, the following commands will work without root privileges. If not (for example, because another user is logged into a tty), systemd will automatically ask you for the root password.
```
This makes sense, however it doesn't make sense that systemd would ALWAYS ask you for a password on ubuntu even if you're the only user logged in. Also, on ubuntu the auth only happens for hibernate, not for poweroff, which is kind of weird. Even though we have the same polkit policy for hibernate as above, it works for me on arch without authentication but doesn't work on ubuntu without authentication. 

Can you please try changing your hibernate policy for to 
```
  implicit any:      yes
  implicit inactive: yes
  implicit active:   yes
```
and then try hibernating again with systemctl and with my PR. If this fixes it then I don't think there is anything we can do from the code side as it is a system configuration issue, where the user decides if hibernating is possible without auth. I'm guessing systemd is more complicated because it's more security conscious compared to the deprecated  apis. If this makes it work then possibly the best we can do is create a FAQ for users when they find that they cannot hibernate on a Linux system that users systemd.

**Comment by sanderjo** [comment]
> Can you please try changing your hibernate policy for to
> 
>   implicit any:      yes
>   implicit inactive: yes
>   implicit active:   yes

Where can I do that? I looked at the structure in /etc/polkit-1, but I'm not sure.

FWIW https://help.ubuntu.com/stable/ubuntu-help/power-hibernate.html says hibernate won't work with btrfs (which I have for /), but that seems untrue, as hibernate does work (after the authentication).

**Comment by sanderjo** [comment]
Thanks to https://askubuntu.com/a/820296 Hibernate is now working on my Ubuntu 17.10 (so despite btrfs)

**Comment by Safihre** [comment]
👍

**Comment by BrendanBall** [comment]
the stackoverflow I found said to edit /usr/share/polkit-1/actions/org.freedesktop.login1.policy and change it there. However in general you should never modify default settings and rather override them as the defaults might be changed back during upgrades. So the link you found looks like the right solution. Do we want to prepare a FAQ to include this for when the PR gets released in v2.2?

Also regarding btrfs, last time I checked it doesn't support swap, unless you create a loopback device to a swapfile which is apparently really slow. Is your whole harddrive btrfs?

**Comment by sanderjo** [comment]
Root is btrfs, home is not:

sander@R540:~$ mount | grep "^/dev" 
/dev/sda2 on / type btrfs (rw,relatime,ssd,space_cache,subvolid=257,subvol=/@)
/dev/sda6 on /home type ext4 (rw,relatime,data=ordered)
sander@R540:~$ 
sander@R540:~$ swapon
NAME      TYPE      SIZE   USED PRIO
/dev/sda5 partition 4,2G 187,6M   -1
sander@R540:~$
